### ISSUE TITLE
Downloading Cache with HI3Mirror Produce Error

### ISSUE THREAD

**Issue by bagusnl** [BuIlDaLiBlE/BetterHI3Launcher#49]
**Describe the bug**
Updating HI3 cache with HI3Mirror as selected mirror produces error

**To Reproduce**
1. Pick HI3Mirror as download mirror
2. Click on 'Download Cache`
3. See error

**Expected behavior**
Cache will be downloaded using HI3Mirror

**Log**
If you haven't closed the launcher yet, add the output of entire log window; otherwise search for the appropriate log in this path: `%userprofile%\AppData\LocalLow\Bp\Better HI3 Launcher`.
```
Connecting to Hi3Mirror...success!
Verifying game cache...
File corrupted: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Data\data\2578608036_E95DB003CD3074A3B63626397BE3B142844FFBB2.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Data\data\2578608104_26FB4A359DCCD1D5FDBBE91C235F62E73FA13154.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Data\data\excel_output_AF3203BC04469C9E085ECE8F3D120EE287D582D8.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Data\data\setting_DFFF24C55C63D77942C4A98C880821DA60302056.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\cgreplay\const_cg44783_truecolor_DA31E9FDD7656581962695310414872D685CD3C0.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.6_figures122822_truecolor_FE5E67F3DBBC17CD360C74C11A7C9E3F2A874C1E.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.6_icons122822_truecolor_389DE830D916E5269E2C53EB8C8C46AA3235BD32.unity3d
Finished verifying files, found corrupted/missing files: 7
ERROR: System.FormatException: Input string was not in a correct format.
   at System.Text.StringBuilder.FormatError()
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at BetterHI3Launcher.MainWindow.<>c__DisplayClass78_4.<<DownloadGameCache>b__7>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at BetterHI3Launcher.MainWindow.<DownloadGameCache>d__78.MoveNext()
```

**Screenshots**

**Additional context**

**Comment by BuIlDaLiBlE** [comment]
Can't reproduce. Can you restart the launcher and try again?

**Comment by bagusnl** [comment]
Here I tried to delete some caches so it triggers again:
```
BetterHI3Launcher v1.3.20220327.0
Working directory: E:\GameData\HonkaiImpact3\SEA\Games\
OS version: Windows 11 Pro (Version 21H2, Build 22000.588)
OS language: en-US
Launcher language: en (autodetect)
Using server: SEA
Using mirror: Hi3Mirror
Checking for game update...
Game version: 5.5.0
Game directory: E:\GameData\HonkaiImpact3\SEA\Games
Connecting to Hi3Mirror...success!
Verifying game cache...
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\5.5_smallicons1061468_truecolor_23F75E5364BB550E16898813DF3B3FCC341801AA.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.5_figures115165_truecolor_7C8054BE496CB9891BBC0366BC83A92C2788420C.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.5_figures122815_truecolor_51EC214DAD194D92651A33224F363796876219DC.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.5_icons115165_truecolor_29D7D516C9E58B2C1599BD8B69B3799176DF5749.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.5_icons122815_truecolor_86D4465F3F883A2FCD5D53C4F64EEA1C55A8656F.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.5_smallicons122815_truecolor_ED94B3D5DF05CA1820E73C7FA22B18270D2A4110.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.6_figures122822_truecolor_FE5E67F3DBBC17CD360C74C11A7C9E3F2A874C1E.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\back_5.6_icons122822_truecolor_389DE830D916E5269E2C53EB8C8C46AA3235BD32.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_figures1061468_truecolor_45FE63620B46972232E2B6C4E402F48D765669A4.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_giftpack302020_truecolor_9D909507B5001EC1E1893A18075B90CC6D69276E.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_giftpack302021_truecolor_203138D74687AEAE63A0DC0D8EC9522001EC8E45.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons1061438_truecolor_739B14CA3BE5E8AF472F47AD07E2A9575782B19F.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons1061468_truecolor_E6AF71DA4E8DEA984F0787FAE7B03ED5B1A522B5.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons122805_truecolor_4E86B041D40DE381D696B5C49E32C8DB95C13DFC.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons122806_truecolor_073B682037492C583CF433C3F1DBDC551B3337A3.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons122807_truecolor_F5E5A61E53785B21F1893EE318EFA7846D827E87.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons122814_truecolor_A8E01CECDF1A9467BC3B053296B8D0F4C47898A4.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons122819_truecolor_93699D42B1E9F35067B96D3B11B813DA6824B370.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons3510_truecolor_24F95BDBB9A668A46B6578F7C83487251F2BB042.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_icons3511_truecolor_55583495B83254475B12F1888B8D6C5A5B12AF04.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_iconscs_truecolor_1E794BCD06186E043EBF1D4E02265909A0213046.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_iconsfx_truecolor_FEAFE3588F2EFE2089E72066A8B79B88E5207F67.unity3d
File missing: C:\Users\bagusnl_reg\AppData\LocalLow\miHoYo\Honkai Impact 3\Resources\event\immediately\material\const_iconsleilv_truecolor_1D4B33B4F3DB8BBBA67C100DDD34E9859D082C3D.unity3d
Finished verifying files, found corrupted/missing files: 23
ERROR: System.FormatException: Input string was not in a correct format.
   at System.Text.StringBuilder.FormatError()
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at BetterHI3Launcher.MainWindow.<>c__DisplayClass78_4.<<DownloadGameCache>b__7>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at BetterHI3Launcher.MainWindow.<DownloadGameCache>d__78.MoveNext()
```
![image](https://user-images.githubusercontent.com/28079733/160292526-1b589ae5-dee8-45ec-995b-46784b0c636f.png)

**Comment by BuIlDaLiBlE** [comment]
Addressed in [v1.3.20220327.1](https://github.com/BuIlDaLiBlE/BetterHI3Launcher/releases/tag/1.3.20220327.1).
